core/src/main/scala/io/projectglow/vcf/VariantContextToInternalRowConverter.scala

```diff
@@ -285,6 +286,48 @@ class VariantContextToInternalRowConverter(
     makeArray(strList, parseFn)
   }
 
+  private def getAnnotationArray(
+      vc: VariantContext,
+      key: String,
+      schema: StructType): Array[GenericInternalRow] = {
+    vc.getAttributeAsStringList(key, "")
+      .asScala
+      .toArray
+      .map { effect =>
+        // Providing a limit to the splitter preserves empty annotations
+        val stringAnnotations =
+          effect.split(AnnotationUtils.annotationDelimiterRegex, schema.size)
+        val convertedAnnotations = stringAnnotations.zip(schema.fields).map {
+          case ("", _) => null
+          case (strAnn, annField) =>
+            annField.dataType match {
+              case ArrayType(StringType, _) =>
+                new GenericArrayData(
+                  strAnn
+                    .split(AnnotationUtils.arrayDelimiter)
+                    .map(UTF8String.fromString(_).asInstanceOf[Any]))
+              case st if st.isInstanceOf[StructType] =>
```

Comment:
This case is not obvious. It's for the multi field annotations? Please document.